Makefile

```diff
@@ -10,8 +10,10 @@ TOOLSDIR := $(CURDIR)/internal/build
 TMPDIR ?= $(CURDIR)/.tmp
 OUTDIR ?= $(TMPDIR)
 
-GOLANGCI_LINT_VERSION ?= v1.55
-REVIVE_VERSION ?= v1.3.7
+# Dynamic version selection based on Go version
+# Format: $(TOOLSDIR)/get_version.sh <go_version> <tool_version1> <tool_version2> ..
+GOLANGCI_LINT_VERSION ?= $(shell $(TOOLSDIR)/get_version.sh 1.20 v1.55 v1.59)
```

Comment:
any particular reason for which we hold back on the lint version? Why `v1.59` when latest is `v1.61.0` ?


[tags](https://github.com/golangci/golangci-lint/tags)

Reply:
> any particular reason for which we hold back on the lint version? Why `v1.59` when latest is `v1.61.0` ?
> 
> [tags](https://github.com/golangci/golangci-lint/tags)

in this branch go1.22 is the latest supported Go. and newer versions of golangci-lint include new checks that require deep changes to the code base, which is out of the scope of this PR